Identity/Identity/Program.cs

```diff
@@ -49,11 +51,35 @@
     })
     .AddGoogle(googleOptions =>
     {
-        googleOptions.ClientId = builder.Configuration["Auth:Google:ClientID"]!;
-        googleOptions.ClientSecret = builder.Configuration["Auth:Google:ClientSecret"]!;
+        googleOptions.ClientId = builder.Configuration["Auth:Google:ClientID"];
+        googleOptions.ClientSecret = builder.Configuration["Auth:Google:ClientSecret"];
         googleOptions.SignInScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+        googleOptions.Events = new OAuthEvents
+        {
+            OnCreatingTicket = async context =>
+            {
+                // Extract the email from the Principal
+                var emailClaim = context.Principal.FindFirst(ClaimTypes.Email);
+                if (emailClaim == null)
+                {
+                    throw new Exception("Email claim not found");
+                }
+                var email = emailClaim.Value;
+
+                // Rest of your code
+                var userService = context.HttpContext.RequestServices.GetRequiredService<UsersService>();
```

Comment:
Here will be runtime error as at this place not in the DB. Check other callbacks maybe there postAuth or something like that appears